lorax/lorax.py

```diff
@@ -216,13 +544,31 @@ def _plot_graph(self, idx, pred_class, score,
 
         plt.show()
 
-    def _build_contrib_df(self, mean_by_trees_list, idx, how):
+    def _build_contrib_df_sample(self, mean_by_trees_list, how):
+        contrib_df = pd.DataFrame(mean_by_trees_list,
+                                  columns=['feature', 'contribution'])
+        contrib_df.set_index('feature', inplace=True)
+
+        # if we're using name patterns, aggregate columns to pattern level,
+        # otherwise, join on column-level statistics (not available for pattern-level)
+        if how == 'patterns':
+            contrib_df = contrib_df.join(self.column_patterns, how='inner')
+            contrib_df = contrib_df.groupby(['name_pattern'])['contribution'].sum().to_frame()
+
+        # sort the resulting dataframe in descending order by contribution
+        contrib_df.sort_values('contribution', ascending=False, inplace=True)
+
+        return contrib_df
+
+    def _build_contrib_df(self, mean_by_trees_list, test_mat, idx, sample, feature_stats, how='features'):
         """
         Build contribution dataframe.
 
         In:
             - mean_by_trees_list (list):
+            - test_mat: The reference test matrix, a dataframe
             - idx: index for example
+            - sample: the row matrix of the sample. Either a idx or sample should be provided
```

Comment:
should include `feature_stats` in the input list

Reply:
hmmm... if I'm reading, correctly, it seems like you'd have to explicitly pass `None` for either `idx` or `sample` here. Could either make it a single required parameter (where you check the type) or default both to `None`, so they're not required/positional arguments.